fv/test-connection/test-connection.go

```diff
@@ -119,7 +140,16 @@ func main() {
 	}
 }
 
-func tryConnect(ipAddress, port, sourcePort, protocol, loopFile string) error {
+func maybeAddInterface(sourceIP string) error {
+	var err error
+	if sourceIP != defaultSourceIP {
+		cmd := exec.Command("ip", "addr", "add", sourceIP+"/32", "dev", "eth0")
```

Comment:
This won't work for IPv6 since you hardcode `/32`

Reply:
Thanks, 46acbf95f898593f230a8cb4762973055aecb9af